schematron/v2.1.0/tests/test_L000_Prelim_Analysis.py

```diff
@@ -0,0 +1,85 @@
+import os
+
+from lxml import etree
+
+from tools.validate_sch import validate_schematron
+from conftest import failures_by_role, AssertFailureRolesMixin, golden_tree, remove_element, SCH_DIR
+
+class TestL000PrelimAnalysis(AssertFailureRolesMixin):
+    schematron = os.path.join(SCH_DIR, 'L000_Prelim_Analysis.sch')
+    golden_file = os.path.join(SCH_DIR, 'golden_files', 'L000_Prelim_Analysis.xml')
+
+    def test_golden_file_is_valid(self):
+        # -- Act
+        failures = validate_schematron(self.schematron, self.golden_file)
+
+        # -- Assert
+        self.assert_failure_roles(failures, {})
+
+    def test_fails_when_measured_scenario_missing(self):
+        # -- Setup
+        tree = golden_tree('L000_Prelim_Analysis')
+
+        # verify it's valid
+        failures = validate_schematron(self.schematron, tree)
+        self.assert_failure_roles(failures, {})
+
+        # remove the measured scenario
+        tree = remove_element(tree, '//auc:Scenario[auc:ScenarioType/auc:CurrentBuilding/auc:CalculationMethod/auc:Measured]')
+        
+        # -- Act 
+        failures = validate_schematron(self.schematron, tree)
+
+        # -- Assert
+        self.assert_failure_roles(failures, {'ERROR': 3})
```

Comment:
Can we be explicit about what the 3 error messages should be?  It would just help when reading through the tests to make sure that the errors are firing that we are expecting to fire.

Reply:
Updated 👍